### PR TITLE
Add Dice similarity

### PR DESCRIPTION
**Pull request by mjste**
## Changes

* Added dice similarity

## Checklist before requesting a review
- [x] Docstrings added/updated in public functions and classes
- [x] Tests added, reasonable test coverage (at least ~90%, `make test-coverage`)
- [x] Sphinx docs added/updated and render properly (`make docs` and see `docs/_build/index.html`)

### CODE REVIEW

File: skfp/distances/dice.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+        Second binary input array or sparse matrix.
+
+    Returns
+    ----------
+    similarity : float
+        Dice similarity between vec_a and vec_b.
+
+    References
+    ----------
+
```
Comment (j-adamczyk) [inline skfp/distances/dice.py:49]: Useful references:
- https://esajournals.onlinelibrary.wiley.com/doi/10.2307/1932409
- https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8026075/#sec014title
- https://pubs.acs.org/doi/10.1021/ci300261r

File: skfp/distances/dice.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    vec_a : {ndarray, sparse matrix}
+        First binary input array or sparse matrix.
+
+    vec_b : {ndarray, sparse matrix}
+        Second binary input array or sparse matrix.
+
+    Returns
+    ----------
+    similarity : float
+        Dice similarity between vec_a and vec_b.
```
Comment (j-adamczyk) [inline skfp/distances/dice.py:45]: Double backticks for vec_a and vec_b

File: skfp/distances/dice.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    Dice similarity for vectors of binary values.
+
+    Computes the Dice similarity for binary data between two input arrays
+    or sparse matrices using the formula:
+
+    .. math::
+
+        sim(vec_a, vec_b) = \\frac{2 |vec_a \\cap vec_b|}{|vec_a| + |vec_b|}
+
+    The calculated similarity falls within the explicit range `[0, 1]`.
```
Comment (j-adamczyk) [inline skfp/distances/dice.py:31]: Double backticks. Also, why "explicit" range? It's just a range

File: skfp/distances/dice.py [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+
+
+@validate_params(
+    {
+        "vec_a": ["array-like", csr_array],
+        "vec_b": ["array-like", csr_array],
+    },
+    prefer_skip_nested_validation=True,
+)
+def dice_binary_similarity(
```
Comment (j-adamczyk) [inline skfp/distances/dice.py:18]: Also add count variant, see formulas here:
- https://jcheminf.biomedcentral.com/articles/10.1186/s13321-015-0069-3/tables/2
- https://jcheminf.biomedcentral.com/articles/10.1186/s13321-015-0069-3

File: skfp/distances/dice.py [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+    if np.sum(vec_a) == 0 == np.sum(vec_b):
+        return 1.0
+
+    if isinstance(vec_a, csr_array) and isinstance(vec_b, csr_array):
+        vec_a_bool = vec_a.astype(bool)
+        vec_b_bool = vec_b.astype(bool)
+
+        intersection: float = vec_a_bool.multiply(vec_b_bool).sum()
+
+        return 2 * intersection / (vec_a_bool.sum() + vec_b_bool.sum())
```
Comment (j-adamczyk) [inline skfp/distances/dice.py:81]: I prefer to avoid empty lines in `if/elif/else`

File: skfp/distances/dice.py [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+
+    Computes the Dice distance for binary data between two input arrays
+    or sparse matrices by subtracting the similarity from 1, using to
+    the formula:
+
+    .. math::
+
+        dist(vec_a, vec_b) = 1 - sim(vec_a, vec_b)
+
+    The calculated distance falls within the range `[0, 1]`.
```
Comment (j-adamczyk) [inline skfp/distances/dice.py:116]: Double backticks

File: skfp/distances/dice.py [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+    vec_a : {ndarray, sparse matrix}
+        First binary input array or sparse matrix.
+
+    vec_b : {ndarray, sparse matrix}
+        Second binary input array or sparse matrix.
+
+    Returns
+    ----------
+    distance : float
+        Dice distance between vec_a and vec_b.
```
Comment (j-adamczyk) [inline skfp/distances/dice.py:130]: Double backticks

**Review by j-adamczyk — CHANGES_REQUESTED** [review]

File: skfp/distances/dice.py (hunk 3 above)
Comment (mjste) [reply skfp/distances/dice.py:31]: Tbh, copy paste of tanimoto. Seemed good enough.
I will fix it.

Comment (j-adamczyk) [reply skfp/distances/dice.py]: Please also fix it in Tanimoto then, I missed it before

**Review by j-adamczyk — APPROVED** [review]

**Pull request merged**